pkg/deploy/dev-workspace/dev_workspace.go

```diff
@@ -62,19 +54,22 @@ func NewDevWorkspaceReconciler() *DevWorkspaceReconciler {
 }
 
 func (d *DevWorkspaceReconciler) Reconcile(ctx *chetypes.DeployContext) (reconcile.Result, bool, error) {
-	if isDevWorkspaceOperatorCSVExists(ctx) {
-		// Do nothing if DevWorkspace has been already deployed via OLM
-		return reconcile.Result{}, true, nil
-	}
-
 	if infrastructure.IsOpenShift() {
-		wtoInstalled, err := isWebTerminalSubscriptionExist(ctx)
-		if err != nil {
+		// Do nothing if explicitly declared not to manage Dev Workspace resources
+		isNoOptDWO, err := isNoOptDWO()
+		if isNoOptDWO {
+			return reconcile.Result{}, true, nil
+		} else if err != nil {
 			return reconcile.Result{Requeue: true}, false, err
 		}
-		if wtoInstalled {
-			// Do nothing if WTO exists since it should bring or embeds DWO
+
+		// Do nothing if Dev Workspace operator has owner (installed via OLM).
+		// In this case Dev Workspace operator resources mustn't be managed by Che operator.
+		isDevWorkspaceOperatorHasOwner, err := isDevWorkspaceOperatorHasOwner(ctx)
```

Comment:
I worry that down the line we may run into similar issues in non-OLM installs if 

1. I manually install DWO to namespace A
2. I install Che to namespace B

From reading the code, it looks like what will happen in this case is that Che find the DWO deployment, see it doesn't have an owner, and try to deploy/update DWO. However, since we're not looking for an install namespace, there's a chance Che will attempt to deploy an second instance of the operator, breaking workspace reconcile. 

This PR does fix the main issue we're facing though, so :+1: to the solution.

Reply:
Yes, your are right. 
There is one assumption behind this.
DWO is always installed into `devworkspace-controller` namespace for non-OLM installs [1]

[1] https://github.com/devfile/devworkspace-operator/blob/main/deploy/deployment/openshift/objects/devworkspace-controller-manager.Deployment.yaml#L8